pkg/controller/atlasdeployment/atlasdeployment_controller.go

```diff
@@ -477,3 +482,24 @@ func (r *AtlasDeploymentReconciler) removeDeletionFinalizer(context context.Cont
 }
 
 type deploymentHandlerFunc func(ctx *workflow.Context, project *mdbv1.AtlasProject, deployment *mdbv1.AtlasDeployment, req reconcile.Request) (workflow.Result, error)
+
+// Parse through tags and verfiy that all keys are unique. Return error on duplicate key.
```

Comment:
This can be easily unit tested. Being part of the spec, I think it makes even more sense.